.github/workflows/publish_pages.js.yml

```diff
@@ -15,24 +15,28 @@ jobs:
   build-and-deploy:
     concurrency: ci-${{ github.ref }} # Recommended if you intend to make multiple deployments in quick succession.
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+         python-version: [3.9]
     steps:
     - uses: actions/checkout@v3
-    - uses: snickerbockers/submodules-init@v4
     - name: Install and Build 🔧 # This example project is built using npm and outputs the result to the 'build' folder. Replace with the commands required to build your project, or remove this step entirely if your site is pre-built.
       run: |
         npm i
         npm run build
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install pyppeteer
+      run: |
+        pip install pyppeteer
+    - name: Run pyppeteer test
+      run: |
+        python pyppeteer_test.py
```

Comment:
The key is this line -- run the python script.  If it fails, the workflow is aborted and we won't deploy to gh-pages.